lib/models/code.ts

```diff
@@ -0,0 +1,20 @@
+import warehouse from 'warehouse';
+import type Hexo from '../hexo';
+import { CodeSchema } from '../types';
+import { join } from 'path';
+
+export = (ctx: Hexo) => {
+  const Code = new warehouse.Schema<CodeSchema>({
+    _id: { type: String, required: true },
+    path: { type: String, required: true },
+    slug: { type: String, required: true },
```

Comment:
Because the same value is already set in `path`, I don't think `slug` is necessary. Though the difference may be slight, reducing the elements in `Schema<CodeSchema>` should have a positive effect on performance. 

However, if we remove the `slug`, it could potentially be a breaking change since there may be users referencing it in custom scripts or etc..., so it might require an announcement.